src/Api/Endpoints/Candidates.php

```diff
@@ -0,0 +1,234 @@
+<?php declare(strict_types=1);
+/**
+ * This implements the Candidates endpoint class
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+namespace LORIS\Api\Endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Api\Endpoint;
+/**
+ * A class for handling the api/v????/candidates endpoint.
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+class Candidates extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+    /**
+     * A cache of the results of the endpoint, so that it doesn't
+     * need to be recalculated for the ETag and handler
+     */
+    protected $cache;
+
+    /**
+     * Permission checks
+     *
+     * @param \User $user The requesting user
+     *
+     * @return boolean true if access is permitted
+     */
+    protected function hasAccess(\User $user)
+    {
+        return (
+            $user->hasPermission('access_all_profiles') ||
+            ($user->hasStudySite() && $user->hasPermission('data_entry'))
+        );
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return array(
+                'GET',
+                'POST',
+               );
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return [
+                "v0.0.1",
+                "v0.0.2",
+                "v0.0.3-dev",
+               ];
+    }
+
+    /**
+     * Handles a request starts with /candidates
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $user = $request->getAttribute('user');
+        if ($user instanceof \LORIS\AnonymousUser) {
+            return new \LORIS\Http\Response\Unauthorized();
+        }
+
+        if (!$this->hasAccess($user)) {
```

Comment:
would it revel that the user is valid compare to the previous error?

Reply:
The previous checks if the user is logged in.
The second check for that user's permissions